src/components/network/network-loader.js

```diff
@@ -19,117 +19,251 @@ genotet.utils.inherit(genotet.NetworkLoader, genotet.ViewLoader);
  * @param {string} fileName Network File name.
- * @param {string} geneRegex Regex for gene selection.
+ * @param {string} inputGenes The input genes.
+ * @param {boolean} isRegex Whether inputGene is regex.
  * @override
  */
-genotet.NetworkLoader.prototype.load = function(fileName, geneRegex) {
-  this.loadNetwork_(fileName, geneRegex);
+genotet.NetworkLoader.prototype.load = function(fileName, inputGenes, isRegex) {
+  var genes = this.prepareGenes_(inputGenes, isRegex);
+  this.data.genes = genes;
+  this.loadNetwork_(fileName, genes);
+};
+
+/**
+ * Prepares gene array for the input gene string and isRegex.
+ * @param {string} inputGenes Input string for genes.
+ * @param {boolean} isRegex If inputGene is regex.
+ * @return {!Array<string>}
+ * @private
+ */
+genotet.NetworkLoader.prototype.prepareGenes_ = function(inputGenes, isRegex) {
+  var genes = [];
+  if (isRegex) {
+    var regex;
+    try {
+      regex = RegExp(inputGenes, 'i');
+    } catch (e) {
+      genotet.error('invalid gene regex', inputGenes);
+      return [];
+    }
+    this.data.networkInfo.nodes.forEach(function(node) {
+      if (node.id.match(regex)) {
+        genes.push(node.id);
+      }
+    });
+  } else {
+    genes = inputGenes.split(/[\s,]+/);
+  }
+  return genes;
+};
+
+/**
+ * Gets the network info.
+ * @param {string} fileName File name of the network.
+ */
+genotet.NetworkLoader.prototype.loadNetworkInfo = function(fileName) {
+  var params = {
+    type: 'network-info',
+    fileName: fileName
+  };
+  this.get(genotet.data.serverURL, params, function(data) {
+    this.data.networkInfo = data;
+    this.data.networkInfo.nodeLabel = {};
+    this.data.networkInfo.isTF = {};
+    this.data.networkInfo.nodes.forEach(function(node) {
+      this.data.networkInfo.nodeLabel[node.id] = node.label;
+      this.data.networkInfo.isTF[node.id] = node.label;
+    }, this);
+    this.signal('infoLoaded');
+  }.bind(this), 'cannot load network info');
 };
 
 /**
  * Implements the network loading ajax call.
- * @param {string} fileName FIle Name of the network.
- * @param {string} geneRegex Regex that selects the gene set.
+ * @param {string} fileName File Name of the network.
+ * @param {!Array<string>} genes Genes that selects the gene set.
  * @private
  */
-genotet.NetworkLoader.prototype.loadNetwork_ = function(fileName,
-                                                        geneRegex) {
+genotet.NetworkLoader.prototype.loadNetwork_ = function(fileName, genes) {
   var params = {
     type: 'network',
     fileName: fileName,
-    geneRegex: geneRegex
+    genes: genes
   };
+
   this.get(genotet.data.serverURL, params, function(data) {
-    // Store the last applied fileName and geneRegex.
+    if (!genes.length) {
+      genotet.warning('input gene not found');
+      return;
+    }
+    // Store the last applied fileName and genes.
     _.extend(data, {
       fileName: fileName,
-      geneRegex: geneRegex
+      genes: genes,
+      edgeIdtoId: {}
```

Comment:
Will you ever be using network.edges array? If not you can convert it to an object. I'm not sure whether it's used elsewhere so maybe it's safer to create a separate id->edge object mapping.